db/seeds.d/16-mail_notifications.rb

```diff
@@ -49,11 +50,14 @@
     :description        => N_('A test message to check the email configuration is working'),
     :mailer             => 'UserMailer',
     :method             => 'tester',
-    :subscription_type  => false
+    :subscriptable      => false,
+    :subscription_type  => nil
   }
 ]
 
 notifications.each do |notification|
-  MailNotification.create(notification) if MailNotification.where(:name => notification[:name]).blank?
+  mail = MailNotification.where(:name => notification[:name]).first_or_initialize(notification)
+  mail.new_record? ? mail.save : mail.assign_attributes(notification)
+  mail.save if mail.subscriptable_changed?
```

Comment:
The error in the test occurred because   `rake db:seed` saved records even though nothing was changed.
The `subscriptable` field is the issue here, so if it been changed, the mail notification record will save.